### ISSUE TITLE
Don't issue network requests during build

### ISSUE THREAD

**Issue by anmonteiro** [paurkedal/confero#1]
My curiosity was piqued by the release of this library to OPAM.

I tried packaging it for Nix, and the first problem I ran into was `wget` being called during `dune build`.

Would it make sense to distribute those artifacts as part of the library distribution itself rather than performing network requests during build?

This is not urgent, as I can work around it, but thought I'd raise.

**Comment by paurkedal** [comment]
I agree that building from a package should not issue a network request. The way I did it with the opam package, was to use the extra-source directive, allowing opam to take care of the downloading, and placing it such that the dune fallback-mode rule did not trigger.  I am not familiar with nix packaging, but maybe there is a similar feature of having multiple sources?  If so, these are the extra-source directives which need to be translated:
```
extra-source "lib/ducet/allkeys.txt" {
  src: "https://www.unicode.org/Public/UCA/15.0.0/allkeys.txt"
  checksum: "sha512=26dda70e65d38afd5da36fce32a89615880a50e10518545d638fc0112aa7816e3c245bd719de7d55dd205c82581415fc04a86f3401f3e91607f0caaf368c0761"
}
extra-source "test/CollationTest.zip" {
  src: "https://www.unicode.org/Public/UCA/15.0.0/CollationTest.zip"
  checksum: "sha512=fdc5a089d15d2202ea496e86baece55cf7cd1afb7a53b3f23e2f3d4506ac7853517142e9159c18ee25c5b3f7b928a109cbd5737fb19fd76cd445178b9117a6c7"
}
```
The latter is only needed if tests are run; I omitted it for the current opam release to avoid the extra download in the common case where tests are not run. There is no `{with-test}` condition available for extra-source, as far as I could tell.

That said, my first though was to include the data files in the source tarball, which would make things simpler for external packaging.  However, this is currently not supported by dune-release.  That can probably be fixed though.
